### PR TITLE
testing/cachefilesd-inotify: new aport

### PR DESCRIPTION
**Pull request by tomalok**
This is a fork of `community/cachefilesd` which has been patched to work with INOTIFY instead of DNOTIFY (which is disabled in most Alpine Linux kernels).

It is hoped that this fork will eventually be brought back into cachefilesd's mainline codebase, but in the meantime, this package should unblock those who might have wanted to use `cachefilesd` with Alpine Linux.

Of course, in order to this to be useful, PR #11762 also needs to be merged.

### CODE REVIEW

**Comment by tomalok** [comment]
Closed #11762 in favor of this new aport.

File: testing/cachefilesd-inotify/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+pkgrel=0
+pkgdesc="Userspace daemon acting as a backend for FS-Cache (uses INOTIFY instead of DNOTIFY)"
+url="https://gitlab.com/tomalok/cachefilesd-inotify"
+arch="all"
+license="GPL"
+depends="attr"
+makedepends="rpm file"
+replaces="cachefilesd"
+options="!check"
+subpackages="$pkgname-doc"
```
Comment (Cogitri) [inline testing/cachefilesd-inotify/APKBUILD]: I guess you should add a `$pkgname-openrc` subpackage for the initd file

File: testing/cachefilesd-inotify/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Contributor: Jake Buchholz <tomalok@gmail.com>
+# Maintainer: Jake Buchholz <tomalok@gmail.com>
+pkgname=cachefilesd-inotify
+pkgver=0.11.0
+pkgrel=0
+pkgdesc="Userspace daemon acting as a backend for FS-Cache (uses INOTIFY instead of DNOTIFY)"
+url="https://gitlab.com/tomalok/cachefilesd-inotify"
+arch="all"
+license="GPL"
```
Comment (Cogitri) [inline testing/cachefilesd-inotify/APKBUILD]: That isn't a valid license identifier as of https://spdx.org/licenses

File: testing/cachefilesd-inotify/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+depends="attr"
+makedepends="rpm file"
+replaces="cachefilesd"
```
Comment (Cogitri) [inline testing/cachefilesd-inotify/APKBUILD]: I think it needs to both replace and provide it

File: testing/cachefilesd-inotify/APKBUILD [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+options="!check"
+subpackages="$pkgname-doc"
+source="https://gitlab.com/tomalok/cachefilesd-inotify/-/archive/$pkgver/$pkgname-$pkgver.tar.bz2
+	cachefilesd.initd"
+builddir="$srcdir"/$pkgname-$pkgver
```
Comment (Cogitri) [inline testing/cachefilesd-inotify/APKBUILD]: No need for this, this is the default already.

File: testing/cachefilesd-inotify/APKBUILD [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+build() {
+	cd "$builddir"
```
Comment (Cogitri) [inline testing/cachefilesd-inotify/APKBUILD]: No need for this, abuild does this by default.

File: testing/cachefilesd-inotify/APKBUILD [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	sed -i "s#/sbin/#/usr/bin/#g" cachefilesd.c
+	sed -i "s#/sbin/#/usr/bin/#g" cachefilesd.service
+	sed -i "s/^secctx/#secctx/g" cachefilesd.conf
+	make CFLAGS="$CFLAGS"
+}
+
+package() {
+	cd "$builddir"
```
Comment (Cogitri) [inline testing/cachefilesd-inotify/APKBUILD]: Same here

File: testing/cachefilesd-inotify/APKBUILD (hunk 1 above)
Comment (tomalok) [reply testing/cachefilesd-inotify/APKBUILD]: adding...

File: testing/cachefilesd-inotify/APKBUILD (hunk 2 above)
Comment (tomalok) [reply testing/cachefilesd-inotify/APKBUILD]: Updated to `"GPL-2.0-or-later"` as the RPM spec file in the source says "GPLv2+"

File: testing/cachefilesd-inotify/APKBUILD (hunk 3 above)
Comment (tomalok) [reply testing/cachefilesd-inotify/APKBUILD]: good catch, updating.

File: testing/cachefilesd-inotify/APKBUILD (hunk 4 above)
Comment (tomalok) [reply testing/cachefilesd-inotify/APKBUILD]: ok.

File: testing/cachefilesd-inotify/APKBUILD (hunk 5 above)
Comment (tomalok) [reply testing/cachefilesd-inotify/APKBUILD]: ok.

File: testing/cachefilesd-inotify/APKBUILD (hunk 6 above)
Comment (tomalok) [reply testing/cachefilesd-inotify/APKBUILD]: ditto.

**Comment by tomalok** [comment]
The maintainers of `cachefilesd` have not gotten back to me with regards to merging into the mainline codebase.  I'm committed to maintaining this fork for as long as necessary, and have been using it for some time now...  Would it be possible to merge directly into `community` instead of `testing`, @Cogitri?

**Comment by Cogitri** [comment]
No, please put it into testing first so we can at least see that it works on all arches and then file a PR to move it to community

File: testing/cachefilesd-inotify/APKBUILD [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+pkgname=cachefilesd-inotify
+pkgver=0.11.0
+pkgrel=0
+pkgdesc="Userspace daemon acting as a backend for FS-Cache (uses INOTIFY instead of DNOTIFY)"
+url="https://gitlab.com/tomalok/cachefilesd-inotify"
+arch="all"
+license="GPL-2.0-or-later"
+depends="attr"
+makedepends="rpm file"
+provides="cachefilesd"
```
Comment (Cogitri) [inline testing/cachefilesd-inotify/APKBUILD]: I think provides needs to be like `provides="cachefilesd=$pkgver-r$pkgrel`

Comment (tomalok) [reply testing/cachefilesd-inotify/APKBUILD]: https://wiki.alpinelinux.org/wiki/APKBUILD_Reference#provides doesn't mention `$pkgrel`, I'm not sure behavior beyond `$pkgver` is defined?  (Releases may not necessarily be kept in sync...)

I've updated to add the `$pkgver`.

**Comment by Cogitri** [comment]
@maxice8 mind taking a look?

File: testing/cachefilesd-inotify/APKBUILD [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+pkgname=cachefilesd-inotify
+pkgver=0.11.0
+pkgrel=0
+pkgdesc="Userspace daemon acting as a backend for FS-Cache (uses INOTIFY instead of DNOTIFY)"
+url="https://gitlab.com/tomalok/cachefilesd-inotify"
+arch="all"
+license="GPL-2.0-or-later"
+depends="attr"
+makedepends="rpm file"
+provides="cachefilesd=$pkgver"
```
Comment (maxice8) [inline testing/cachefilesd-inotify/APKBUILD]: provides="cachefilesd=$pkgver-r$pkgrel"

**Comment by tomalok** [comment]
@Cogitri @maxice8 - updated provides as directed.

**Comment by Cogitri** [comment]
Thanks for the ping :)

**Pull request closed without merging**